src/client/main.ts

```diff
@@ -232,21 +231,25 @@ if (channelPromise) {
 /**
  * Adds a listener for incoming Intents from the Agent.
  */
-export function addIntentListener(intent: string, handler: (context: Context) => void): IntentListener {
+export function addIntentListener(intent: string, handler: (context: Context) => any): IntentListener {
     const listener = {
         intent,
         handler,
         unsubscribe: () => {
-            const index: number = contextListeners.indexOf(listener);
+            const index: number = intentListeners.indexOf(listener);
 
             if (index >= 0) {
-                contextListeners.splice(index, 1);
+                intentListeners.splice(index, 1);
             }
 
             return index >= 0;
         }
     };
     intentListeners.push(listener);
+
+    tryServiceDispatch(APIFromClientTopic.INTENT_LISTENER_READY, {
+        intent
```

Comment:
Feel like this would be more readable and consistent with similar calls as a one-liner

Reply:
done